p2p/pex/pex_reactor.go

```diff
@@ -449,7 +467,19 @@ func (r *PEXReactor) ensurePeers() {
 
 	// Dial picked addresses
 	for _, addr := range toDial {
-		go r.dialPeer(addr)
+		go func(addr *p2p.NetAddress) {
+			err := r.dialPeer(addr)
+			if err != nil {
+				switch err.(type) {
+				case errMaxAttemptsToDial:
```

Comment:
we can put multiple cases on the same line if the handling is the same:

`case errMaxAttemptsToDial, errTooEarlyToDial:`

Also, why not move all this into the dialPeer method?

Reply:
I needed this method to indicate whenever dialing was successful or not. I can refactor it to return `(success bool)` and move logging into the method itself